src/EFCore.Cosmos/Query/Internal/CosmosShapedQueryCompilingExpressionVisitor.ParameterInliner.cs

```diff
@@ -80,48 +80,29 @@ protected override Expression VisitExtension(Expression expression)
                 }
 
                 // Inlines array parameter of full-text functions, transforming FullTextContainsAll(x, @keywordsArray) to FullTextContainsAll(x, keyword1, keyword2)) 
+                // we do this for FullTextContainsAll, FullTextContainsAny and FullTextScore
                 case SqlFunctionExpression
                 {
-                    Name: "FullTextContainsAny" or "FullTextContainsAll",
+                    Name: string name,
+                    IsScoringFunction: bool scoringFunction,
                     Arguments: [var property, SqlParameterExpression { TypeMapping: { ElementTypeMapping: var elementTypeMapping }, Type: Type type } keywords]
                 } fullTextContainsAllAnyFunction
-                when type == typeof(string[]):
+                when (name == "FullTextContainsAny" || name == "FullTextContainsAll" || name == "FullTextScore") && type == typeof(string[]):
```

Comment:
Embrace the new stuff:

```suggestion
                when (name is "FullTextContainsAny" or "FullTextContainsAll" or "FullTextScore") && type == typeof(string[]):
```